tests/TestKdbx4.cpp

```diff
@@ -510,3 +511,30 @@ void TestKdbx4Format::testCustomData()
     QCOMPARE(newEntry->customData()->value(customDataKey1), customData1);
     QCOMPARE(newEntry->customData()->value(customDataKey2), customData2);
 }
+
+void TestKdbx4Format::testMultiFactorHeaderRead()
+{
+    QString filename = QString(KEEPASSX_TEST_DATA_DIR).append("/MultiFactorPasswordOnly.kdbx");
+
+    auto key = QSharedPointer<CompositeKey>::create();
+    auto passwordKey = QSharedPointer<PasswordKey>::create();
+    passwordKey->setPassword(QByteArray::fromStdString("somepassword"));
```

Comment:
This KDBX file created with keepasspy doesn't actually use SHA256("somepassword") as its key. It uses a random key stored encrypted in the header, and SHA256("somepassword") can decrypt that key part.

In other words, this test shows the whole flow working end-to-end.